dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

```diff
@@ -1,1005 +1,1005 @@
-        if (interfaceClass != null && !interfaceClass.isInterface()) {
-            throw new IllegalStateException("The interface class " + interfaceClass + " is not a interface!");
-        }
-        this.interfaceClass = interfaceClass;
-        setInterface(interfaceClass == null ? null : interfaceClass.getName());
-    }
-
-    public void setInterface(String interfaceName) {
-        this.interfaceName = interfaceName;
-        if (StringUtils.isEmpty(id)) {
-            id = interfaceName;
-        }
-    }
-
-    public T getRef() {
-        return ref;
-    }
-
-    public void setRef(T ref) {
-        this.ref = ref;
-    }
-
-    @Parameter(excluded = true)
-    public String getPath() {
-        return path;
-    }
-
-    public void setPath(String path) {
-        checkPathName(Constants.PATH_KEY, path);
-        this.path = path;
-    }
-
-    public List<MethodConfig> getMethods() {
-        return methods;
-    }
-
-    // ======== Deprecated ========
-
-    @SuppressWarnings("unchecked")
-    public void setMethods(List<? extends MethodConfig> methods) {
-        this.methods = (List<MethodConfig>) methods;
-    }
-
-    public ProviderConfig getProvider() {
-        return provider;
-    }
-
-    public void setProvider(ProviderConfig provider) {
-        ConfigManager.getInstance().addProvider(provider);
-        this.provider = provider;
-    }
-
-    @Parameter(excluded = true)
-    public String getProviderIds() {
-        return providerIds;
-    }
-
-    public void setProviderIds(String providerIds) {
-        this.providerIds = providerIds;
-    }
-
-    public String getGeneric() {
-        return generic;
-    }
-
-    public void setGeneric(String generic) {
-        if (StringUtils.isEmpty(generic)) {
-            return;
-        }
-        if (ProtocolUtils.isGeneric(generic)) {
-            this.generic = generic;
-        } else {
-            throw new IllegalArgumentException("Unsupported generic type " + generic);
-        }
-    }
-
-    @Override
-    public void setMock(Boolean mock) {
-        throw new IllegalArgumentException("mock doesn't support on provider side");
-    }
-
-    @Override
-    public void setMock(String mock) {
-        throw new IllegalArgumentException("mock doesn't support on provider side");
-    }
-
-    public List<URL> getExportedUrls() {
-        return urls;
-    }
-
-    /**
-     * @deprecated Replace to getProtocols()
-     */
-    @Deprecated
-    public List<ProviderConfig> getProviders() {
-        return convertProtocolToProvider(protocols);
-    }
-
-    /**
-     * @deprecated Replace to setProtocols()
-     */
-    @Deprecated
-    public void setProviders(List<ProviderConfig> providers) {
-        this.protocols = convertProviderToProtocol(providers);
-    }
-
-    @Override
-    @Parameter(excluded = true)
-    public String getPrefix() {
-        return Constants.DUBBO + ".service." + interfaceName;
-    }
-}
+/*
```

Comment:
what happened to this file ? pls make it clear what you have chenged to this file

Reply:
sorry, I had make some conflict....I will make a new pr later